tests/integration/rdb.tcl

```diff
@@ -259,6 +272,21 @@ start_server {overrides {save ""}} {
                 fail "COW info wasn't reported"
             }
 
+            # assert that $current_processed_keys is not greater than keys on start.
+            assert_morethan_equal $keys_on_bgsave_start $current_processed_keys
+  
+            # wait to see that $current_processed_keys value updated (as long as the child is in progress)
+            wait_for_condition 80 100 {
+                [s rdb_bgsave_in_progress] == 0 ||
+                [s current_processed_keys] > $current_processed_keys
+            } else {
+                if {$::verbose} {
+                    puts "current_processed_keys info on fail: [s current_processed_keys]"
+                    puts [exec tail -n 100 < [srv 0 stdout]]
+                }
+                fail "current_processed_keys info wasn't reported"
+            }
```

Comment:
i don't think we want to add another "wait_for" loop for this.
the one we have above can validate both info fields.
or alternatively, the one we have about can remain a "wait_for" on COW, then here all we need is an `assert_morethan`.
i.e. once we found the COW report increased, we can just validate that the key count increased too (will make the code simpler to read)

Reply:
Done